### DISCUSSION TITLE
Filter on an multiple choices value (stored as array)

### DISCUSSION THREAD

**Discussion by ajira86** [directus/directus#10078] [Archive]
I try to get a collection with a value contained in an array with something like

> /items/products?filter=[color][_contains]=red

Where color can contains something like ["red", "green"] but I always have an internal server error code even I use any methods to filter or group or anything else on this kind of field

> error select "products"."id", "products"."color" from "products" where ("products"."color" like $1) order by "products"."id" asc limit $2 - operator does not exist: json ~~ unknown

Where I am wrong ?

**Comment by rijkvanzanten** [comment]
`_contains` searches for a substring in a piece of text and can't be used on a JSON value. JSON values are blobs in a database that can't be searched through (yet). There's an open discussion discussing how to add dedicated JSON filters here: https://github.com/directus/directus/discussions/7277

**Reply by ajira86** [reply]
So, should I use CSV data type instead ?

**Reply by rijkvanzanten** [reply]
CSVs are stored as a string type, so those will support the string filter operators. The real solution is the final implementation of what's being discussed in #7277, though that (obv) doesn't exist yet
